16 Lazy Loading/src/index.tsx

```diff
@@ -2,19 +2,41 @@ import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 import { Router, Route, IndexRoute, hashHistory } from 'react-router';
 import App from './components/app.tsx';
-import AboutPage from './components/about/aboutPage';
-import MembersPage from './components/members/membersPage';
-import MemberPage from './components/member/memberPage';
 
+//Loading single component in one chunk
+const lazyLoadAboutComponent = () => {
+  return {
+      getComponent: (location, callback)=> {
+        require.ensure([], require => {
+          callback(null, require('./components/about/aboutPage')["default"]);
+        }, 'AboutPage');
+      }
+    }
+};
+
+//Loading group of components in one chunk
```

Comment:
The trick was in request both components in the array of require.ensure `require.ensure(['./components/member/memberPage', './components/members/membersPage']` and then call to callback function depends member route.

Other way to write this:

```
<Router history={hashHistory}>
    <Route  path="/" component= {App} >
      <IndexRoute {...lazyLoadAboutComponent()} />
      <Route path="/about" {...lazyLoadAboutComponent()} />

      <Route path="/members" getComponent={(location, callback) => {
        require.ensure(['./components/member/memberPage', './components/members/membersPage'], require => {
          callback(null, require('./components/members/membersPage')["default"]);
        }, 'MemberComponents');
      }} />

      <Route path="/member" getComponent={(location, callback) => {
        require.ensure(['./components/member/memberPage', './components/members/membersPage'], require => {
          callback(null, require('./components/member/memberPage')["default"]);
        }, 'MemberComponents');
      }} />

      <Route path="/memberEdit/:id" getComponent={(location, callback) => {
        require.ensure(['./components/member/memberPage', './components/members/membersPage'], require => {
          callback(null, require('./components/member/memberPage')["default"]);
        }, 'MemberComponents');
      }} />
    </Route>
  </Router>
```